font-patcher

```diff
@@ -904,6 +904,21 @@ class font_patcher:
             self.sourceFont.hhea_ascent = self.sourceFont.os2_winascent
             self.sourceFont.hhea_descent = -self.sourceFont.os2_windescent
 
+    def add_glyphrefs_to_essential(self, unicode):
+        self.essential.add(unicode)
+        # According to fontforge spec, altuni is either None or a tuple of tuples
+        if self.sourceFont[unicode].altuni is not None:
+            for r in self.sourceFont[unicode].altuni:
+                # If alternate unicode already exists in self.essential,
+                # that means it has gone through this function before.
+                # Therefore we skip it to avoid infinite loop.
+                # A unicode value of -1 basically means unused and is also worth skipping.
+                if r[0] not in self.essential and r[0] >= 0:
```

Comment:
I like list comprehension, but that is maybe a question of personal style

```diff

-        if self.sourceFont[unicode].altuni is not None:
+        altuni = self.sourceFont[unicode].altuni
+        if altuni is not None:
-            for r in self.sourceFont[unicode].altuni:
+            for altcode in [ v for v, s, r in altuni if v >= 0 ]:
                 # If alternate unicode already exists in self.essential,
                 # that means it has gone through this function before.
                 # Therefore we skip it to avoid infinite loop.
                 # A unicode value of -1 basically means unused and is also worth skipping.
-                if r[0] not in self.essential and r[0] >= 0:
-                   self.add_glyphrefs_to_essential(r[0])
+                if altcode not in self.essential:
+                   self.add_glyphrefs_to_essential(altcode)
```

At least maybe add explanation for the `[0]`, by adding a line with
`# altuni is tuple of (unicode-value, variation-selector, reserved-field)`

Reply:
I'll try this right away, thanks for the recommendation! In all honestly I'm not very knowledgeable with Python so I actually have never heard of list comprehension, until today, that is.